src/Symfony/Component/Clock/Tests/ClockTest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Clock\Clock;
+use Symfony\Component\Clock\MockClock;
+use Symfony\Component\Clock\NativeClock;
+
+class ClockTest extends TestCase
+{
+    public function testClock()
+    {
+        $this->assertInstanceOf(\DateTimeImmutable::class, Clock::now());
+        $this->assertInstanceOf(NativeClock::class, Clock::get());
+
+        $clock = new MockClock();
+
+        $this->assertSame([123, $clock], Clock::with($clock, fn ($arg) => [$arg, Clock::get()], 123));
+        $this->assertInstanceOf(NativeClock::class, Clock::get());
```

Comment:
You should add a test on `Clock::get($mockclock)`.

Reply:
test added